datalad/support/gitrepo.py

```diff
@@ -3424,16 +3429,35 @@ def save_(self, message=None, paths=None, _status=None, **kwargs):
             if (props.get('state', None) in ('modified', 'untracked') and
                 not (f in to_add_submodules or f in to_stage_submodules))}
         if to_add:
+            compat_config = \
+                self.config.obtain("datalad.save.windows-compat-warning")
+            to_add, problems = self._check_for_win_compat(to_add, compat_config)
```

Comment:
I think the encapsulation isnt optimal here. `compat_config` is localized to the helper function, but must be passed in from the outside.